### DISCUSSION TITLE
Can't install on windows

### DISCUSSION THREAD

**Discussion by bali182** [openai/whisper#1942] [Q&A] [answered]
I'm sorry if it's a noob question, but I've never written a line of python in my life, so I'm kinda stumped figuring out what's going on.

I have installed on a Windows 11 machine this (checking using `-V`):

- Python 3.12.1
- pip 23.3.2
- ffmpeg 6.1.1 (using Chocolately as described - `choco install ffmpeg`)
- setuptools-rust (`pip install setuptools-rust` - no idea what this is but saw it on similar questions)

When I run `pip install` for either the github source or the one in the package manager, I get this:

> PS C:\> pip install -U openai-whisper
> Collecting openai-whisper
>   Using cached openai-whisper-20231117.tar.gz (798 kB)
>   Installing build dependencies ... done
>   Getting requirements to build wheel ... done
>   Preparing metadata (pyproject.toml) ... done
> Collecting numba (from openai-whisper)
>   Using cached numba-0.58.1.tar.gz (2.6 MB)
>   Installing build dependencies ... done
>   Getting requirements to build wheel ... error
>   error: subprocess-exited-with-error
> 
>   × Getting requirements to build wheel did not run successfully.
>   │ exit code: 1
>   ╰─> [21 lines of output]
>       Traceback (most recent call last):
>         File "C:\Users\Balazs\AppData\Local\Programs\Python\Python312\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 353, in <module>
>           main()
>         File "C:\Users\Balazs\AppData\Local\Programs\Python\Python312\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 335, in main
>           json_out['return_val'] = hook(**hook_input['kwargs'])
>                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>         File "C:\Users\Balazs\AppData\Local\Programs\Python\Python312\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 118, in get_requires_for_build_wheel
>           return hook(config_settings)
>                  ^^^^^^^^^^^^^^^^^^^^^
>         File "C:\Users\Balazs\AppData\Local\Temp\pip-build-env-1zhmapgk\overlay\Lib\site-packages\setuptools\build_meta.py", line 325, in get_requires_for_build_wheel
>           return self._get_build_requires(config_settings, requirements=['wheel'])
>                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>         File "C:\Users\Balazs\AppData\Local\Temp\pip-build-env-1zhmapgk\overlay\Lib\site-packages\setuptools\build_meta.py", line 295, in _get_build_requires
>           self.run_setup()
>         File "C:\Users\Balazs\AppData\Local\Temp\pip-build-env-1zhmapgk\overlay\Lib\site-packages\setuptools\build_meta.py", line 480, in run_setup
>           super(_BuildMetaLegacyBackend, self).run_setup(setup_script=setup_script)
>         File "C:\Users\Balazs\AppData\Local\Temp\pip-build-env-1zhmapgk\overlay\Lib\site-packages\setuptools\build_meta.py", line 311, in run_setup
>           exec(code, locals())
>         File "<string>", line 51, in <module>
>         File "<string>", line 48, in _guard_py_ver
>       RuntimeError: Cannot install on Python version 3.12.1; only versions >=3.8,<3.12 are supported.
>       [end of output]
> 
>   note: This error originates from a subprocess, and is likely not a problem with pip.
> error: subprocess-exited-with-error
> 
> × Getting requirements to build wheel did not run successfully.
> │ exit code: 1
> ╰─> See above for output.
> 
> note: This error originates from a subprocess, and is likely not a problem with pip.

I restarted the terminal after installing all the dependencies and running it with elevated privileges. Any ideas what might be the issue? Is this meant to run on Windows?

**Comment by phineas-pta** [comment, marked as answer]
> RuntimeError: Cannot install on Python version 3.12.1; only versions >=3.8,<3.12 are supported.

**Reply by AMDphreak** [reply]
Is there any way you guys could update the package install script to reflect this answer, to warn users, to point to this error report, or ultimately to fix it? Just something to keep people updated on the progress would be epic. It's 2025 June and this problem is either still here or back. Not sure which.

**Reply by phineas-pta** [reply]
im not working nor affiliated with the devs so it's useless to tell me

otherwise, if u have problem, share the full logs so other can help

**Comment by bali182** [comment]
> > RuntimeError: Cannot install on Python version 3.12.1; only versions >=3.8,<3.12 are supported.

Thank you very much! Didn't expect breaking changes between 3 patch versions so thought the newer the better. Didn't even read it after the stack traces. I learn something new every day :)
